lib/gitlab/composer/repository/project_repository.rb

```diff
@@ -0,0 +1,84 @@
+require 'digest/crc32'
+
+module Gitlab
+  module Composer
+    module Repository
+      class ProjectRepository < ::Composer::Repository::WritableHashRepository
+
+        # Initializes the project repository.
+        # @param [Composer::Json::JsonFile] repository_file The repository json file
+        def initialize(repository_file)
+          unless repository_file
+            raise ::Composer::ArgumentError,
```

Comment:
since we are subclassing ::Composer::Repository::WritableHashRepository I used the ::Composer errors for compatibility.